tests/integration-tests/tests/common/utils.py

```diff
@@ -90,11 +90,53 @@ def retrieve_latest_ami(region, os, ami_type="official", architecture="x86_64"):
         sys.exit(1)
 
 
+def retrieve_pcluster_ami_without_standard_naming(region, os, version="2.8.1", architecture="x86_64"):
+    try:
+        client = boto3.client("ec2", region_name=region)
+        ami_name = f"ami-for-testing-pcluster-version-validation-without-standard-naming-{version}-{os}"
+        amis = client.describe_images(
+            Filters=[{"Name": "name", "Values": [ami_name]}, {"Name": "architecture", "Values": [architecture]}],
+        ).get("Images", [])
+        if amis:
+            # If the AMI is found, return the AMI
+            return amis[0]["ImageId"]
+        else:
+            # If no AMI is found, copy from an official AMI
+            official_ami_name = "aws-parallelcluster-{version}-{ami_name}".format(
+                version=version, ami_name=OS_TO_PCLUSTER_AMI_NAME_OWNER_MAP.get(os).get("name")
+            )
+            official_amis = client.describe_images(
+                Filters=[
+                    {"Name": "name", "Values": [official_ami_name]},
+                    {"Name": "architecture", "Values": [architecture]},
+                ],
+                Owners=OS_TO_PCLUSTER_AMI_NAME_OWNER_MAP.get(os).get("owners"),
+            ).get("Images", [])
+            return client.copy_image(
+                Description="This AMI is a copy from an official AMI but uses a different naming. "
+                "It is used to bypass the AMI's name validation of pcluster version "
+                "to test the validation in Cookbook.",
+                Name=ami_name,
+                SourceImageId=official_amis[0]["ImageId"],
+                SourceRegion=region,
+            ).get("ImageId")
+
+    except ClientError as e:
+        LOGGER.critical(e.response.get("Error").get("Message"))
+        sys.exit(1)
```

Comment:
you don't want to exit the process in the tests but always raise exception so that pytest executes the cleanup handlers. you could also avoid catching an exception, the framework will log it automatically

Reply:
I will change it